src/shogun/machine/KernelMachine.cpp

```diff
@@ -333,64 +337,62 @@ SGVector<float64_t> CKernelMachine::apply_get_outputs(CFeatures* data)
 		}
 		else
 		{
-			// TODO: port to use OpenMP backend instead of pthread
-#ifdef HAVE_PTHREAD
-			int32_t num_threads=parallel->get_num_threads();
-#else
-			int32_t num_threads=1;
-#endif
-			ASSERT(num_threads>0)
-
-			if (num_threads < 2)
+			int32_t num_threads;
+			int64_t step;
+#pragma omp parallel shared(num_threads, step)
 			{
-				S_THREAD_PARAM_KERNEL_MACHINE params;
-				params.kernel_machine=this;
-				params.result = output.vector;
-				params.start=0;
-				params.end=num_vectors;
-				params.verbose=true;
-				params.indices = NULL;
-				params.indices_len = 0;
-				apply_helper((void*) &params);
-			}
-#ifdef HAVE_PTHREAD
-			else
-			{
-				pthread_t* threads = SG_MALLOC(pthread_t, num_threads-1);
-				S_THREAD_PARAM_KERNEL_MACHINE* params = SG_MALLOC(S_THREAD_PARAM_KERNEL_MACHINE, num_threads);
-				int32_t step= num_vectors/num_threads;
-
-				int32_t t;
 
-				for (t=0; t<num_threads-1; t++)
+#ifdef HAVE_OPENMP
+#pragma opm single
```

Comment:
typo.... due to this as it was mentioned by @karlnapf in https://github.com/shogun-toolbox/shogun/pull/3738#issuecomment-289268657 that be great that our unit test somehow tests both multi and single threaded scenario...